src/node/blockstorage.h

```diff
@@ -216,8 +216,13 @@ class BlockManager
     //! Returns last CBlockIndex* that is a checkpoint
     const CBlockIndex* GetLastCheckpoint(const CCheckpointData& data) EXCLUSIVE_LOCKS_REQUIRED(cs_main);
 
+    //! Returns 'lower_block' if all blocks up to that point are available in disk, otherwise returns the first block that is not pruned in the upper-lower range.
+    //! If 'lower_block=nullptr': verify blocks availability up to the genesis block.
+    //! Returns nullptr if 'lower_block' is not an 'upper_block' ancestry.
+    const CBlockIndex* IsBlockDataAvailable(const CBlockIndex& upper_block LIFETIMEBOUND, const CBlockIndex* lower_block=nullptr) EXCLUSIVE_LOCKS_REQUIRED(::cs_main);
```

Comment:
> I don't think this is possible to trigger with the way the function is used, but if upper_block already has NO_DATA, it would return that block, maybe nullptr would be more natural? (also applies to GetFirstStoredBlock).

yeah, that is something that already exists. Good eye.
It shouldn't be possible to trigger as `upper_block` is always equal to the chain tip and, as far as I know, we never prune the tip. But.. a bug is a bug, better to fix it now.